api/v1beta1/machine_types.go

```diff
@@ -404,11 +406,13 @@ type MachineSpec struct {
 	// This field will be set by the actuators and consumed by higher level entities like autoscaler that will
 	// be interfacing with cluster-api as generic provider.
 	// +optional
+	// +kubebuilder:validation:MaxLength=253
```

Comment:
Provider IDs can be much longer than this, worst case on Azure is somewhere just over 400 chars. Suggest 512 for this

Reply:
Thx. Will use that

Reply:
Also changed for providerIDList for MP